EpikCord/client/http_client.py

```diff
@@ -0,0 +1,193 @@
+        ...
+
+    def __eq__(self, other: MockBucket):
+        if isinstance(other, MockBucket):
+            return True
+        return False
+
+class Bucket:
+    def __init__(self, *, bucket_hash: str):
+        """
+        Parameters
+        ----------
+        bucket_hash: str
+            The bucket hash of the bucket provided by Discord.
+
+        Attributes
+        ----------
+        bucket_hash: str
+            The bucket hash of the bucket provided by Discord.
+        event: asyncio.Event
+            The event that is used to wait for the bucket to be set.
+        """
+        self.hash: str = bucket_hash
+        self.event: asyncio.Event = asyncio.Event()
+        self.event.set()
+
+    async def wait(self):
+        """
+        Waits for the bucket to be set.
+        """
+        await self.event.wait()
+
+    def set(self):
+        """
+        Sets the bucket.
+        """
+        self.event.set()
+
+    def clear(self):
+        """
+        Clears the bucket.
+        """
+        self.event.clear()
+
+
+    def __eq__(self, other: Bucket):
+        if isinstance(other, Bucket):
+            return self.hash == other.hash
+        return False
+
+class TopLevelBucket(Bucket):
+    def __init__(self, *, major_parameters: Dict[str, Any]):
+        """
+        Parameters
+        ----------
+        internal_hash: str
+            The internal hash of the bucket.
+        external_hash: str
+            The external hash of the bucket.
+        major_parameters: Dict[str, Any]
+            The major parameters for this Bucket
+
+        Attributes
+        ----------
+        internal_hash: str
+            The internal hash of the bucket.
+        external_hash: str
+            The external hash of the bucket.
+        major_parameters: Dict[str, Any]
+            The major parameters for this Bucket
+        """
+        super().__init__(internal_hash=internal_hash, external_hash=external_hash)
+        self.major_parameters: Dict[str, Any] = major_parameters
+
+    def __eq__(self, other: TopLevelBucket):
+        if isinstance(other, TopLevelBucket):
+            return self.major_parameters == other.major_parameters
+        return False
+
+class HTTPClient:
+    def __init__(self, token: str, *, version: int = 10):
+        self.token: str = token
+        self.version: int = version
+        self.session: aiohttp.ClientSession = aiohttp.ClientSession(headers={
+            "Authorization": f"Bot {self.token}"
+        })
+        self.buckets: Dict[str, Bucket] = {}
+        self.global_event: asyncio.Event = asyncio.Event()
+        self.global_event.set()
+
+    async def ws_connect(self, url: str) -> aiohttp.ClientWebSocketResponse:
+        return await self.session.ws_connect(url)
+
+    async def request(self, method: str, url: str, *args,  discord: bool = True, channel_id: int = 0, guild_id: int = 0, webhook_id: int = 0, webhook_token: str = "None", **kwargs) -> Optional[aiohttp.ClientResponse]:
```

Comment:
This function as way to maybe methods. 
Some kind of request data object could be passed.

The discord boolean is also not very explicit.